### PR TITLE
add elastic training blog

### PR DESCRIPTION
**Pull request by zw0610**
https://github.com/kubeflow/community/issues/500

@kubeflow/wg-training-leads

### CODE REVIEW

File: _posts/2021-03-15-elastic-training.md
Comment (xieydd) [inline _posts/2021-03-15-elastic-training.md:5]: Practise -> Practice

File: _posts/2021-03-15-elastic-training.md
Comment (xieydd) [inline _posts/2021-03-15-elastic-training.md:10]: With increasing of the dataset and deep learning models ..?

File: _posts/2021-03-15-elastic-training.md
Comment (xieydd) [inline _posts/2021-03-15-elastic-training.md:18]: `training`? change ' to `

**Review by xieydd — CHANGES_REQUESTED** [review]

**Review by terrytangyuan — APPROVED** [review]
Looks great! Great work on supporting elastic training on MPI operator! @zw0610 @gaocegege 

/lgtm

**Comment by gaocegege** [comment]
@kubeflow/wg-training-leads 

@andreyvelich 

/lgtm

File: _posts/2021-03-15-elastic-training.md
Comment (zw0610) [reply _posts/2021-03-15-elastic-training.md]: fixed

File: _posts/2021-03-15-elastic-training.md
Comment (zw0610) [reply _posts/2021-03-15-elastic-training.md:18]: single quotation seems fine as it's not referring code script.

**Comment by zw0610** [comment]
/hold cancel

rendering issue fixed

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [inline _posts/2021-03-15-elastic-training.md:40]:
```suggestion
3. Training on Cloud: there is a type of resource on the cloud that is called "spot" or "preemptible" instances; it comes with unexpected low price tags but may be retrieved after guaranteed hour expires.
```

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [inline _posts/2021-03-15-elastic-training.md:68]: Should we try to be consistent between graph numeration and these steps ?

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [inline _posts/2021-03-15-elastic-training.md:60]: Is there any link that we can add here for the `discover_hosts.sh` script ?

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [inline _posts/2021-03-15-elastic-training.md:114]: Maybe we can use `kubectl patch` to be more explicit here, what do you think ?

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [inline _posts/2021-03-15-elastic-training.md:153]: Should we add link here? 
```suggestion
As we can see, Elastic Horovod on MPI-Operator now supports tweaking worker replicas dynamically. As a future work, we aim to support  [`Horizontal Pod Autoscaler`](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) to MPIJob as well as other features like designated worker deletion.
```

**Review by andreyvelich — COMMENTED** [review]
Thank you for the great Blog @zw0610 @gaocegege !
I left few comments

File: _posts/2021-03-15-elastic-training.md
Comment (zw0610) [reply _posts/2021-03-15-elastic-training.md]: good idea. I updated the script in the new commit.

File: _posts/2021-03-15-elastic-training.md
Comment (zw0610) [reply _posts/2021-03-15-elastic-training.md]: This example (https://horovod.readthedocs.io/en/stable/elastic_include.html#running-with-horovodrun) includes what the `discover_hosts.sh` is expected to return.

**Comment by jbottum** [comment]
Nit - I would consider removing "like precedent cloud native adoptions" in final paragraph.

File: _posts/2021-03-15-elastic-training.md
Comment (zw0610) [reply _posts/2021-03-15-elastic-training.md]: Since we do not want to re-iterate the mpi-operator proposal in this blog, I changed the numeration to bullet so readers should be not confused. What do you think?

**Comment by zw0610** [comment]
> Nit - I would consider removing "like precedent cloud native adoptions" in final paragraph.

sounds reasonable. While there are some precedents like that, it's not always the case.

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [reply _posts/2021-03-15-elastic-training.md]: @zw0610 It's good. Let's include it in the blog.

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [reply _posts/2021-03-15-elastic-training.md]: Thank you, I think it is better since it will not confuse reader.

**Comment by gaocegege** [comment]
Please take a look, thanks for your time :smile:

@andreyvelich

File: _posts/2021-03-15-elastic-training.md
Comment (andreyvelich) [inline _posts/2021-03-15-elastic-training.md:60]:
```suggestion
When launching an elastic training job, `horovodrun` requires a `discover_hosts.sh` script to detect available hosts and slots in real time. (In the following section, we refer this script as `discover_hosts.sh`. Nevertheless the script needs not to be named as `discover_hosts.sh`. An example of `discover_hosts.sh` can be found [here](https://horovod.readthedocs.io/en/stable/elastic_include.html#running-with-horovodrun).
```

**Review by andreyvelich — COMMENTED** [review]
From the technical side looks good, but I think it would be great if we could have another look on the text style from @jbottum.

File: _posts/2021-03-15-elastic-training.md
Comment (zw0610) [reply _posts/2021-03-15-elastic-training.md:60]: There is a parentheses before "In the following section", which makes a pair with the one at the end of the paragraph.
Are you suggesting remove this pair of parentheses?

Comment (andreyvelich) [reply _posts/2021-03-15-elastic-training.md:60]: Oh, just missed that.
Yes, I think we can remove it.

**Comment by gaocegege** [comment]
cc @tgaddair

**Comment by tgaddair** [comment]
Awesome stuff, @gaocegege!  Great to see Elastic Horovod in Kubeflow :)

**Comment by gaocegege** [comment]
ping @theadactyl

**Comment by theadactyl** [comment]
@gaocegege thanks for the ping! Just confirming that the primary content is re: fully OSS tech. Is that correct?

**Comment by gaocegege** [comment]
Yeah, the feature is merged into kubeflow/mpi-operator.

**Comment by theadactyl** [comment]
Great, thanks @gaocegege! 
/approved for my part
Do we need lgtm from Josh for copy?

**Comment by jbottum** [comment]
/lgtm 

@zw0610 Nice job!   Although some of the sentence structure and word flow might be a little unusual, I believe the message is conveyed.  Also, since I do not have the technical depth to validate the examples, I will rely on the Kubeflow Training WG approvals, which have been provided above.  Thank you for this post!

**Pull request merged**
